controllers/cloud.redhat.com/namespacepool_controller.go

```diff
@@ -54,46 +50,58 @@ type NamespacePoolReconciler struct {
 
 func (r *NamespacePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	pool := crd.NamespacePool{}
+
 	if err := r.Client.Get(ctx, req.NamespacedName, &pool); err != nil {
-		r.Log.Error(err, "Error retrieving namespace pool")
-		return ctrl.Result{}, err
+		r.Log.Error(err, fmt.Sprintf("Cannot retrieve [%s] pool resource", pool.Name))
+		return ctrl.Result{Requeue: true}, err
```

Comment:
If we attempt to retrieve the pool resource but there is an error and we don't requeue, it will proceed without the pool object. I figured a requeue would be good so that we can attempt another shot at retrieving it prior to proceeding with the following login in the reconciler.

Reply:
The same would go for retrieving a list of namespace in the pool. There is some logic that follows so that we can get an accurate reading of namespace count across the `env-status`.